src/OpenTelemetry.SemanticConventions/Resource/ResourceSemanticConventions.cs

```diff
@@ -0,0 +1,714 @@
+// <copyright file="ResourceSemanticConventions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+
+// <auto-generated> This file has been auto generated from buildscripts/semantic-convention/templates/SemanticConventions.cs.j2</auto-generated>
+namespace OpenTelemetry.SemanticConventions.Resource
+{
+    /// <summary>
+    /// Constants for semantic attribute names outlined by the OpenTelemetry specifications.
+    /// <see href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/README.md"/>.
+    /// </summary>
+    public static class ResourceSemanticConventions
+    {
+        /// <summary>
+        /// Name of the cloud provider.
+        /// </summary>
+        public const string AttributeCloudProvider = "cloud.provider";
```

Comment:
What about an inner class `Attribute` and having these generated under it?
So we'd access it through:

`ResourceSemanticConventions.Attribute.CloudProvider`

Reply:
I like the idea. Let's see what the others say then I can change.